homeassistant/components/trafikverket_train/coordinator.py

```diff
@@ -86,16 +83,35 @@ def __init__(
         self._train_api = TrafikverketTrain(
             async_get_clientsession(hass), self.config_entry.data[CONF_API_KEY]
         )
-        self.from_station: StationInfoModel = from_station
-        self.to_station: StationInfoModel = to_station
+        self.from_station: StationInfoModel | None = None
+        self.to_station: StationInfoModel | None = None
         self._time: time | None = dt_util.parse_time(self.config_entry.data[CONF_TIME])
         self._weekdays: list[str] = self.config_entry.data[CONF_WEEKDAY]
         self._filter_product: str | None = self.config_entry.options.get(
             CONF_FILTER_PRODUCT
         )
 
+    async def _async_setup(self) -> None:
+        """Initiate stations."""
+        try:
+            self.to_station = await self._train_api.async_get_train_station(
+                self.config_entry.data[CONF_TO]
+            )
+            self.from_station = await self._train_api.async_get_train_station(
+                self.config_entry.data[CONF_FROM]
+            )
+        except InvalidAuthentication as error:
+            raise ConfigEntryAuthFailed from error
+        except (NoTrainStationFound, MultipleTrainStationsFound) as error:
+            raise UpdateFailed(
+                f"Problem when trying station {self.config_entry.data[CONF_FROM]} to"
+                f" {self.config_entry.data[CONF_TO]}. Error: {error} "
+            ) from error
+
     async def _async_update_data(self) -> TrainData:
         """Fetch data from Trafikverket."""
+        if TYPE_CHECKING:
+            assert self.from_station and self.to_station
```

Comment:
Theoretically, since the async_setup is always run before hand, we could just overwrite the types of from station and to station